server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -1280,7 +1284,11 @@ public void onFailure(Exception e) {
                     }
                 });
             } else {
-                finalizeSnapshotEntry(snapshot, metadata, repositoryData);
+                if (newFinalization) {
+                    finalizeSnapshotEntry(snapshot, metadata, repositoryData);
+                } else {
+                    runNextQueuedOperation(repositoryData, repoName, false);
+                }
```

Comment:
Happy to discuss whether this is the right fix. It seems safe to me. But I am also running this change in a loop to see if there is undesirable side effect.